src/Symfony/Component/Security/Core/Authentication/Provider/DaoAuthenticationProvider.php

```diff
@@ -51,7 +51,12 @@ public function __construct(UserProviderInterface $userProvider, UserCheckerInte
      */
     protected function checkAuthentication(UserInterface $user, UsernamePasswordToken $token)
     {
+        if ($user instanceof PasswordAuthenticatedUserInterface) {
+            throw new \LogicException(sprintf('Class "%s" must implement "%s" for using password-based authentication.',get_debug_type($user), PasswordAuthenticatedUserInterface::class));
+        }
+
         $currentUser = $token->getUser();
+
         if ($currentUser instanceof UserInterface) {
```

Comment:
this should also check `$currentUser instanceof PasswordAuthenticatedUserInterface`

Reply:
done